CMakeLists.txt

```diff
@@ -314,22 +324,49 @@ if(WITH_OTLP)
     endif()
   endif()
   include(CMakeDependentOption)
-  if(WITH_OTLP_HTTP OR (NOT DEFINED WITH_OTLP_HTTP AND NOT DEFINED
-                                                       CACHE{WITH_OTLP_HTTP}))
-    find_package(CURL)
-  endif()
 
   cmake_dependent_option(
     WITH_OTLP_GRPC "Whether to include the OTLP gRPC exporter in the SDK" ON
     "gRPC_FOUND" OFF)
-  cmake_dependent_option(
-    WITH_OTLP_HTTP "Whether to include the OTLP http exporter in the SDK" ON
-    "CURL_FOUND" OFF)
 
   message(STATUS "PROTOBUF_PROTOC_EXECUTABLE=${PROTOBUF_PROTOC_EXECUTABLE}")
   include(cmake/opentelemetry-proto.cmake)
 endif()
 
+#
+# Do we need HTTP CLIENT CURL ?
+#
+
+if(WITH_OTLP_HTTP
+   OR WITH_ELASTICSEARCH
+   OR WITH_JAEGER
+   OR WITH_ZIPKIN
+   OR BUILD_W3CTRACECONTEXT_TEST
+   OR WITH_EXAMPLES_HTTP)
+  set(WITH_HTTP_CLIENT_CURL ON)
+else()
+  set(WITH_HTTP_CLIENT_CURL OFF)
+endif()
+
+#
+# Do we need CURL ?
+#
+
+if(WITH_HTTP_CLIENT_CURL)
+  set(USE_CURL ON)
```

Comment:
Do we need `USE_CURL`, or just `WITH_HTTP_CLIENT_CURL` is enough ?

Reply:
The two are identical because only WITH_HTTP_CLIENT_CURL requires CURL. Simplified the makefile.